### PR TITLE
Added patch for CVE-2023-46136 to python-werkzeug

### PR DESCRIPTION
**Pull request by nisamson**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrades and adds patch to python-werkzeug to fix CVE-2023-46136

###### Change Log  
Upgrades python-werkzeug to 2.3.7
Patches CVE-2023-46136 in python-werkzeug

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-46136

###### Test Methodology

- Pipeline build id: 451961 Unified-Buddy-Build

### CODE REVIEW

**Comment by PawelWMS** [comment]
@nisamson, just a heads-up: all PRs against `fasttrack/2.0` should be rebased, since we finished a monthly update, which also updated `fasttrack/2.0`. Rebasing (or merging) the latest state of the branch will re-trigger the PR check and allow us to validate that the latest state of the repo still builds correctly with the changes from the PRs.

File: SPECS/python-werkzeug/python-werkzeug.spec [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-Version:        2.2.3
+Version:        2.3.7
 Release:        1%{?dist}
 License:        BSD
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Development/Languages/Python
 URL:            https://github.com/pallets/werkzeug
 Source0:        https://github.com/pallets/werkzeug/archive/%{version}.tar.gz#/werkzeug-%{version}.tar.gz
+#Source0:               werkzeug-%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/python-werkzeug/python-werkzeug.spec:11]: Clean-up:

```suggestion
```

File: SPECS/python-werkzeug/python-werkzeug.spec [hunk 2]
```diff
@@ -33,22 +40,37 @@ Werkzeug started as simple collection of various utilities for WSGI applications
...
+# This is a known limitation of tox-current-env. See https://pypi.org/project/tox-current-env/ under "Do not rely on virtualenv details"
+# for more information.
+tox -q --recreate -e %{toxenv}
+
+%files -n python3-werkzeug -f %{pyproject_files}
 %defattr(-,root,root)
-%license LICENSE.rst
-%{python3_sitelib}/*
+%doc README.rst
+%doc CHANGES.rst
```
Comment (PawelWMS) [inline SPECS/python-werkzeug/python-werkzeug.spec:65]: We still want a `%license` macro installing some file explaining the licensing for the package. If the source no longer has anything suitable for that purpose (sometimes licensing info is included in a readme file and that's also OK), then please create a `LICENSE.PTR` file locally and explain the licensing there in a sentence or two. Search for `LICENSE.PTR` in other specs to get an example but usually we end up going through some of the source's code or documentation, find licensing information there and then mention how we found the license in that new file.

Comment (nisamson) [reply SPECS/python-werkzeug/python-werkzeug.spec:66]: Misread the documentation for the `%{pyproject_files}` macro. Will re-add.

**Review by sameluch — APPROVED** [review]
Looks Good!

File: SPECS/python-werkzeug/python-werkzeug.spec [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 Release:        1%{?dist}
 License:        BSD
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Development/Languages/Python
 URL:            https://github.com/pallets/werkzeug
 Source0:        https://github.com/pallets/werkzeug/archive/%{version}.tar.gz#/werkzeug-%{version}.tar.gz
+Patch0:         0001-enable-tests-in-rpm-env.patch
+# Necessary because RPM build environment creates specific scenario that violates a core 
+# assumption the testing infrastructure of werkzeug makes
```
Comment (PawelWMS) [inline SPECS/python-werkzeug/python-werkzeug.spec:13]: I'm fine with a surgical disablement of the tests. I'd only ask to expand the comment a bit, to explain what the "specific scenario" and the "core assumption" are. This way, if we make any changes to the build environment in the future, it's going to be easier to tell, if we can try enable these tests.

Comment (nisamson) [reply SPECS/python-werkzeug/python-werkzeug.spec]: Done

File: SPECS/python-werkzeug/python-werkzeug.spec [hunk 4]
```diff
@@ -33,22 +48,34 @@ Werkzeug started as simple collection of various utilities for WSGI applications
...
 
 %install
-%py3_install
+%pyproject_install
+%pyproject_save_files werkzeug
 
 %check
-pip3 install pytest hypothesis
-LANG=en_US.UTF-8 PYTHONPATH=./  python3 setup.py test
+pip3 install tox==4.6.3 tox-current-env
```
Comment (PawelWMS) [inline SPECS/python-werkzeug/python-werkzeug.spec:62]: Btw. why do we need to pin this to an older version? I'm not saying it's wrong, but let's leave a comment.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**